lib/mongoid/extensions/range.rb

```diff
@@ -59,15 +59,14 @@ def demongoize(object)
         # @example Mongoize the object.
         #   Range.mongoize(1..3)
         #
-        # @param [ Range ] object The object to mongoize.
+        # @param [ Object ] object The object to mongoize.
         #
-        # @return [ Hash ] The object mongoized.
+        # @return [ Hash | nil ] The object mongoized or nil.
         def mongoize(object)
+          return if object.nil?
           case object
-          when NilClass then nil
-          when String then object
           when Hash then __mongoize_hash__(object)
-          else __mongoize_range__(object)
+          when Range then __mongoize_range__(object)
```

Comment:
Arguably an Array or Set with two elements could be converted to range. Perhaps that's a stretch.